R/zchunk_LA102.ag_LDS_R_C_GLU.R

```diff
@@ -33,49 +35,88 @@ module_aglu_LA102.ag_LDS_R_C_GLU_DISABLED <- function(command, ...) {
+    # and convert units for each:
+    # harvested area table, L100.LDS_ag_HA_ha
+    # agricultural production table, L100.LDS_ag_prod_t
     #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # printlog( "Adding GCAM region and commodity info to LDS tables" )
+    # printlog( "Collapsing production and harvested area data to GCAM regions and commodities")
+    # printlog( "Converting GTAP mass to Mt and area to thousand km2 (billion m2, or bm2)")
+
+    # Take the harvested area table, L100.LDS_ag_HA_ha
+    L100.LDS_ag_HA_ha %>%
+      # append GCAM region information from the iso table:
+      left_join_error_no_match(iso_GCAM_regID, by = c("iso")) %>%
+      # remove the country_name and GCAM3 columns added from the iso table:
+      select(-country_name, -region_GCAM3) %>%
+      # append GCAM_commodity information from the FAO table using a left_join to preserve NA values as in old system:
+      left_join(., FAO_ag_items_PRODSTAT, by = c("GTAP_crop")) %>%
+      # select only the relevant columns; iso and GTAP_crop are dropped because they are ommitted from the aggregation:
+      ungroup () %>% select(GCAM_region_ID, GCAM_commodity,GLU, value ) %>%
+      # group by GCAM region, GCAM commodity, and GLU for the aggregation:
+      group_by(GCAM_region_ID, GCAM_commodity, GLU) %>%
+      # aggregate to the GCAM region, commodity, GLU level:
+      summarise_if(is.numeric, sum) %>%
+      # convert units from hectares (ha) to thou km^2 (=billion m^2, bm2):
+      mutate(value=value*CONV_HA_BM2)  %>%
+      # omit na values, since they do not appear in the original table:
+      na.omit()->
+      # store in the final, labeled table for harvested area (HA) in units bm2 at the region-commodity-glu level:
+      L102.ag_HA_bm2_R_C_GLU
+
+    # Take the production table, L100.LDS_ag_HA_ha
+    L100.LDS_ag_prod_t %>%
+      # append GCAM region information from the iso table:
+      left_join_error_no_match(iso_GCAM_regID, by = c("iso")) %>%
+      # remove the country_name and GCAM3 columns added from the iso table:
+      select(-country_name, -region_GCAM3) %>%
+      # append GCAM_commodity information from the FAO table using a left_join to preserve NA values as in old system:
+      left_join(., FAO_ag_items_PRODSTAT, by = c("GTAP_crop")) %>%
+      # select only the relevant columns; iso and GTAP_crop are dropped because they are ommitted from the aggregation:
+      ungroup () %>% select(GCAM_region_ID, GCAM_commodity,GLU, value ) %>%
+      # group by GCAM region, GCAM commodity, and GLU for the aggregation:
+      group_by(GCAM_region_ID, GCAM_commodity, GLU) %>%
+      # aggregate to the GCAM region, commodity, GLU level:
+      summarise_if(is.numeric, sum) %>%
```

Comment:
Why `summarise_if`? Are the `L100.LDS_ag_HA_ha` and `L100.LDS_ag_prod_t` datasets coming in wide format?